contrib/win32/win32compat/misc.c

```diff
@@ -1344,29 +1368,44 @@ get_others_file_permissions(wchar_t * file_name)
 
 		current_aceHeader = (PACE_HEADER)current_ace;
 		/* only interested in Allow ACE */
-		if (current_aceHeader->AceType != ACCESS_ALLOWED_ACE_TYPE)
+		if (!(current_aceHeader->AceType == ACCESS_ALLOWED_ACE_TYPE || 
+		    current_aceHeader->AceType == ACCESS_DENIED_ACE_TYPE))
 			continue;
 
 		PACCESS_ALLOWED_ACE pAllowedAce = (PACCESS_ALLOWED_ACE)current_ace;
```

Comment:
if current_aceHeader is of Denied_ACE_Type, does this statement "PACCESS_ALLOWED_ACE pAllowedAce = (PACCESS_ALLOWED_ACE)current_ace" success?
Suggest to this line:
if(current_aceHeader->AceType == ACCESS_ALLOWED_ACE_TYPE)
{
    pAllowedAce = (PACCESS_ALLOWED_ACE)current_ace;
   current_trustee_sid = &(pAllowedAce->SidStart);
}
else if(current_aceHeader->AceType == ACCESS_DENIED_ACE_TYPE)
{
     pDeniedAce = (PACCESS_DENIEDED_ACE)current_ace;
    current_trustee_sid = &(pDeniedAce->SidStart);
}

Reply:
PACCESS_ALLOWED_ACE, PACCESS_DENIED_ACE are same.. I have tested this.. Just to minimize the code, I didn't have this if/else loop. if you want to be more readable then I can add.

here is the definition.

typedef struct _ACCESS_ALLOWED_ACE {
    ACE_HEADER Header;
    ACCESS_MASK Mask;
    DWORD SidStart;
} ACCESS_ALLOWED_ACE;

typedef ACCESS_ALLOWED_ACE *PACCESS_ALLOWED_ACE;

typedef struct _ACCESS_DENIED_ACE {
    ACE_HEADER Header;
    ACCESS_MASK Mask;
    DWORD SidStart;
} ACCESS_DENIED_ACE;
typedef ACCESS_DENIED_ACE *PACCESS_DENIED_ACE;

Reply:
Please add a comment about PACCESS_ALLOWED_ACE, PACCESS_DENIED_ACE  being the same for code clarify.